packages/flutterfire_starter/hooks/pre_gen.dart

```diff
@@ -12,12 +15,40 @@ Future<void> run(HookContext context) async {
   }
 }
 
-Future<ProcessResult> _generateApp(HookContext context) async {
+void _validatePluginList(HookContext context) {
+  final varsPlugins =
+      ((context.vars['plugins'] as List<dynamic>).cast<String>())
+          .map((e) => e.split(' ')[0])
+          .toList();
+  final setPlugins = varsPlugins.toSet().toList();
+  if (setPlugins.length != varsPlugins.length) {
+    context.logger.err(
+      'You have duplicate plugins in your plugins list. Please remove them.',
+    );
+    exit(1);
+  }
+}
+
+// Inject the array of plugins into the context
+void _injectInContext(HookContext context) {
+  // Default values
+  context.vars = <String, dynamic>{
+    'analyticswithgorouter': false,
+    'analyticswithnavigator': false,
+    ...context.vars
+  };
+  final varsPlugins = (context.vars['plugins'] as List<dynamic>).cast<String>();
+  for (final element in varsPlugins) {
+    context.vars[element.replaceAll(' ', '').toLowerCase()] = true;
+  }
+}
+
+Future<void> _generateApp(HookContext context) async {
   context.logger.info('Running flutter create...');
   final appName = context.vars['name'] as String;
   final appDescription = context.vars['description'] as String;
   final nameOrg = context.vars['org'] as String;
-  return Process.run('flutter', [
+  await Process.run('flutter', [
```

Comment:
Not sure the exception would bubble up from this process. Might be worth adding something like:

```dart
final runScript = await Process.run('flutter', [
    'create',
    appName,
    '-t',
    'skeleton',
    '--description',
    appDescription,
    '--org',
    nameOrg,
  ]);

if (runScript.exitCode != 0) {
   throw Exception(runScript.stderr);
 }
````